### DISCUSSION TITLE
Cargo is not applicable to the toolchain

### DISCUSSION THREAD

**Discussion by thatmagicalcat** [Rahix/avr-hal#409] [Q&A] [answered]
I ran `cargo generate --git https://github.com/Rahix/avr-hal-template.git` to create a new project
and then i ran `cargo build` command and it gave this error
```
$ cargo build
error: the 'cargo' binary, normally provided by the 'cargo' component, is not applicable to the 'nightly-2022-07-10-x86_64-unknown-linux-gnu' toolchain
```
not only for `cargo build` but running any cargo command gives the same error

**Comment by Rahix** [comment, marked as answer]
Try reinstalling the toolchain with rustup, e.g.

```bash
rustup toolchain uninstall nightly-2022-07-10
rustup toolchain install nightly-2022-07-10
```

**Comment by thatmagicalcat** [comment]
It worked thanks
